redpanda/cloud/controlplane.go

```diff
@@ -102,6 +102,24 @@ func (cpCl *ControlPlaneClientSet) ResourceGroupForName(ctx context.Context, nam
 	return nil, fmt.Errorf("resource group %s not found", name)
 }
 
+// ResourceGroupForIDOrName gets the resource group for a given ID or name
+// and handles the error if the returned resource group is nil.
+func (cpCl *ControlPlaneClientSet) ResourceGroupForIDOrName(ctx context.Context, id string, name string) (*controlplanev1beta2.ResourceGroup, error) {
+	if id == "" && name == "" {
+		return nil, fmt.Errorf("resource group must have id or name")
+	}
+	if id != "" && name != "" {
+		return nil, fmt.Errorf("resource group must only have one of id or name")
+	}
+	if id != "" {
+		return cpCl.ResourceGroupForID(ctx, id)
+	}
+	if name != "" {
+		return cpCl.ResourceGroupForName(ctx, name)
+	}
```

Comment:
This should be happening in a valdiator on the datasource schema. See examples here for validating against other elements within a validator

https://github.com/redpanda-data/terraform-provider-redpanda/blob/main/redpanda/validators/cloud_provider.go

Reply:
Updated code to support any combination of ID, name, both, or neither. Similar to other Terraform data sources that allow combining filters.

Reply:
I still feel for UX reasons a validator would be preferable. Users should fail at the plan stage with an incorrectly configured datasource. This will fail only at the apply stage if left blank.

Reply:
@gene-redpanda can you explain what you mean? This currently does fail at the plan stage if it matches zero or more than one resource group(s):
<img width="587" alt="image" src="https://github.com/user-attachments/assets/3f7f114d-0ec1-41a8-8224-018903a8dc8f">

Reply:
Ok, valid point, plan does call read. I should have specified before API call. My preference is that we give them an error message that clearly points to a resolution path. This technically does that, it's just not explicit that the issue is one of ID or name not being set.

Reply:
@r-vasquez What's your thought on this one, do you feel like the error message is sufficient here?

Reply:
> This technically does that, it's just not explicit that the issue is one of ID or name not being set.

This will actually work if no ID or name is set, as long as you have exactly one resource group.

Reply:
👍